### PR TITLE
Refactor(eos_validate_state): Add warning for ansible test mode

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Indicate that only `use_anta: true` behavior will be kept in 5.0.0

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes

After too long searching, added a temporary action plugin to emit a warning. We will remove it when we remove the ansible-test before 5.0.0. This plugin is meant to be used only for this warning

## How to test

In the molecule scenario for validate_state, set `use_anta` to `false` and run. The scenario will fail because it is not meant for this mode, but you will see the emitted warning

Screenshot 2024-07-02 at 09 12 57

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ title: Ansible Collection Role eos_validate_state
 
 # eos_validate_state
 
+!!! warning
+    Starting AVD 5.0.0, the only behavior for `eos_validate_state` will be the current one when using `use_anta: true`.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_validate_state/README.md]:
```suggestion
    Starting AVD 5.0.0, `eos_validate_state` will use ANTA as backend, which will change the default behavior (same as setting `use_anta: true` in older versions).
```

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_validate_state/README.md]: similar for error messages if you agree.

File: ansible_collections/arista/avd/plugins/modules/emit_warning.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - Emit a warning in a task.
+options:
+  message:
+    description: The warning message to emit.
+    type: str
+"""
+
+EXAMPLES = r"""
+---
+- name: Enit a warning
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/modules/emit_warning.py]:
```suggestion
- name: Emit a warning
```

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/ansible_tests.yml [hunk 3]
```diff
@@ -5,6 +5,17 @@
...
+- name: Generate variables for testing
+  arista.avd.emit_warning:
+    message: |-
+      Starting AVD 5.0.0, `eos_validate_state` will use ANTA as backend, which will change the default behavior (same as setting `use_anta: true` in older versions).
+      Please review the changes described on the ANTA integration page on https://avd.arista.com.
+  delegate_to: localhost
+  check_mode: false
+  changed_when: false
+  run_once: true
+  tags:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/ansible_tests.yml:17]:
```suggestion
  when: use_anta is not arista.avd.defined(true)
  tags:
```

File: ansible_collections/arista/avd/docs/release-notes/4.x.x.md [hunk 4]
```diff
@@ -28,6 +28,11 @@ For these reasons, the route-map was removed as a bug fix.
 
 To keep the broken route-map, `structured_config` can be used.
 
+### Deprecations
+
+- Starting AVD 5.0.0, `eos_validate_state` will use ANTA as backend, which will change the default behavior (same as setting `use_anta: true` in older versions).
+  Please review the changes described on the  [ANTA integration page](./anta_integration.md)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/4.x.x.md]:
```suggestion
  Please review the changes described on the [ANTA integration page](./anta_integration.md)
```

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/ansible_tests.yml [hunk 5]
```diff
@@ -5,6 +5,17 @@
 #####################################
 ##  Run ansible eos_validate_state  ##
 #####################################
+- name: Generate variables for testing
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/ansible_tests.yml]:
```suggestion
- name: Warning when not using 'use_anta: true'
```

File: ansible_collections/arista/avd/plugins/modules/_emit_warning.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+DOCUMENTATION = r"""
+---
+module: _emit_warning
+version_added: "4.9.0"
+author: EMEA AS Team (@aristanetworks)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/modules/_emit_warning.py]:
```suggestion
author: Arista Networks (@aristanetworks)
```

File: ansible_collections/arista/avd/docs/plugins/Modules_and_action_plugins/_emit_warning.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+# This title is used for search results
+title: arista.avd._emit_warning
+---
+<!--
+  ~ Copyright (c) 2023-2024 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# _emit_warning
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/plugins/Modules_and_action_plugins/_emit_warning.md:11]: add it to semver

**Review by carlbuchmann — APPROVED** [review]
LGTM - tested in ATD and the deprecation warning was properly displayed when not using ANTA mode!

image

**Pull request merged**